samples/net/sockets/mtls_http_get/server/src/main.rs

```diff
@@ -0,0 +1,83 @@
+use rustls::RootCertStore;
+use rustls::server::AllowAnyAuthenticatedClient;
+use rustls_pemfile::{certs, rsa_private_keys};
+use std::io;
+use std::sync::Arc;
+use tokio::io::{copy, sink, AsyncWriteExt};
+use tokio::net::TcpListener;
+use tokio_rustls::rustls::{self, Certificate, PrivateKey};
+use tokio_rustls::TlsAcceptor;
+
+fn load_certs() -> io::Result<Vec<Certificate>> {
+    static CERT: &[u8] = include_bytes!("../server.crt");
+    certs(&mut CERT.to_owned().as_slice())
+        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid cert"))
+        .map(|mut certs| certs.drain(..).map(Certificate).collect())
+}
+
+fn load_keys() -> io::Result<Vec<PrivateKey>> {
+    static KEY: &[u8] = include_bytes!("../server.key");
+    rsa_private_keys(&mut KEY.to_owned().as_slice())
+        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid key"))
+        .map(|mut keys| keys.drain(..).map(PrivateKey).collect())
+}
+
+fn load_root_store() -> io::Result<RootCertStore> {
+    static CACERT: &[u8] = include_bytes!("../ca.crt");
+    let certs = certs(&mut CACERT.to_owned().as_slice())
+        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid ca cert"))?;
+    let mut root = RootCertStore::empty();
+    for crt in certs {
+        root.add(&Certificate(crt))
+            .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid ca cert"))?;
+    }
+    Ok(root)
+}
+
+#[tokio::main]
+async fn main() -> io::Result<()> {
+    let certs = load_certs()?;
+    let mut keys = load_keys()?;
+    let root_store = load_root_store()?;
+
+    let config = rustls::ServerConfig::builder()
```

Comment:
At some point, we should try to figure out if the TLS maximum record size can be configured in rustls. We will have problems right now if any request ends up larger than the 7k configured for the maximum record size on the client.